integration-tests/test/contracts.spec.ts

```diff
@@ -0,0 +1,202 @@
+import {BigNumber, Contract, ContractFactory, utils, Wallet} from 'ethers';
+import {ethers} from 'hardhat';
+import {solidity} from 'ethereum-waffle';
+import chai, {expect} from 'chai';
+import {UniswapV3Deployer} from 'uniswap-v3-deploy-plugin/dist/deployer/UniswapV3Deployer';
+import {abi as FACTORY_ABI} from '@uniswap/v3-core/artifacts/contracts/UniswapV3Factory.sol/UniswapV3Factory.json';
+import {abi as POOL_ABI} from '@uniswap/v3-core/artifacts/contracts/UniswapV3Pool.sol/UniswapV3Pool.json';
+
+import {OptimismEnv} from './shared/env';
+import {FeeAmount, TICK_SPACINGS} from '@uniswap/v3-sdk';
+
+chai.use(solidity);
+
+export const getMinTick = (tickSpacing: number) => Math.ceil(-887272 / tickSpacing) * tickSpacing;
```

Comment:
Where do these functions come from?

Reply:
Why exactly are these exported?

Reply:
I thought they might be useful in other tests. They come from the Uniswap test suite. I'll unexport and add some comments around where I got them.